### ISSUE TITLE
Consider reordering members so private fields are at the top

### ISSUE THREAD

**Issue by tannergooding** [novelrt/NovelRT#57]
Currently, the header files are declaring fields at the bottom of the class (which I believe is "standard" C++ practice).

This frequently leads to code like:
```cpp
class MyClass {
public:
    inline int getValue() const;

private:
    int _value;
};

inline int MyClass::getValue() const {
    return _value;
}
```

Reordering these such that fields are at the top could simplify this greatly:
```cpp
class MyClass {
private:
    int _value;
public:
    inline int getValue() const {
        return _value;
    }
};
```

**Comment by tannergooding** [comment]
One example is in `GeoVector`, where almost half the lines are redeclaring the template and method signature: https://github.com/novelrt/NovelRT/blob/master/src/GeoVector.h#L62

The ordering also makes it easier to see the state/layout of a given type; without first needing to find the end of the class.

**Comment by RubyNova** [comment]
Resolved by #86 .
